client/asset/dcr/externaltx.go

```diff
@@ -127,12 +123,12 @@ func (dcr *ExchangeWallet) txBlockFromCache(ctx context.Context, tx *externalTx)
 		return nil, nil
 	}
 
-	txBlockStillValid, err := dcr.isMainchainBlock(ctx, tx.block)
```

Comment:
We need to review all the previous calls to `isMainchainBlock`, and the new callers of `blockHeader` with the new bool arg because stake validation and main chain were being combined.

In places where we're doing reorg stuff or main chain ancestor identification, the stakeholder validation is wrong to consider.

In places like `isOutputSpent`, we *may* want to consider if the block's regular tree is invalidated.

Things have been conflated, in part because the previous name `isMainchainBlock` was not accurate.